### PR TITLE
Fixes #1761  Super admin tool to remove bullmq jobs and repeatable jobs 

### LINKED ISSUE #1761: Super admin tool to remove bullmq jobs and repeatable jobs

### PR DESCRIPTION
**Pull request by tonylee80**
add endpoint removebotidjobsfromqueue
add remove Job ID from queue to super admin page
update removeBullMQJobByKey to filter for jobs p.id matching the bot id instead of partial matching p.key

### Summary
🧪🤖🗑️

This pull request adds a new feature for super admins to remove cron jobs by bot id from the queue. It implements a new API endpoint, a new UI component, and some tests for both. It also refactors the `removeBullMQJobByKey` function to use the `id` property of the bot object and to avoid race conditions.

> _Super admins can now clear the queue_
> _Of jobs that belong to a bot id_
> _They use `removeBullMQJobByKey`_
> _And a form with a button and input_
> _To delete cron jobs with a success id_

### Walkthrough
*  Add a new feature to remove bot id jobs from the queue ([link](https://github.com/medplum/medplum/pull/1814/files?diff=unified&w=0#diff-e9fb492809a57eb817e3eb5d9910df8223e6dd212b6f10930b4c09ba1ad82064R141-R151), [link](https://github.com/medplum/medplum/pull/1814/files?diff=unified&w=0#diff-cb04462fdffdc85a5b88a42ae35750a7596f503b5aec6f59495352d10a59c307R173-R195), [link](https://github.com/medplum/medplum/pull/1814/files?diff=unified&w=0#diff-2d51264bfa2bbbde1add9123c15880bfd4b95b624d2add2c7e2ce87d79ec25edL228-R234))
  * Create a new section in the `SuperAdminPage` component to render a form for entering the bot id and a button for submitting the request ([link](https://github.com/medplum/medplum/pull/1814/files?diff=unified&w=0#diff-e9fb492809a57eb817e3eb5d9910df8223e6dd212b6f10930b4c09ba1ad82064R141-R151))
  * Define a new API endpoint in the `super.ts` module to handle the request and validate the bot id and the super admin permission ([link](https://github.com/medplum/medplum/pull/1814/files?diff=unified&w=0#diff-cb04462fdffdc85a5b88a42ae35750a7596f503b5aec6f59495352d10a59c307R173-R195))
  * Modify the `removeBullMQJobByKey` function in the `cron.ts` module to use the id property instead of the key property to filter the jobs and to use a for loop to await the removal of each job ([link](https://github.com/medplum/medplum/pull/1814/files?diff=unified&w=0#diff-2d51264bfa2bbbde1add9123c15880bfd4b95b624d2add2c7e2ce87d79ec25edL228-R234))
* Add a function to handle the form submission and call the API endpoint and show a notification based on the response ([link](https://github.com/medplum/medplum/pull/1814/files?diff=unified&w=0#diff-e9fb492809a57eb817e3eb5d9910df8223e6dd212b6f10930b4c09ba1ad82064R45-R51))
* Add a test case for the new feature in the `SuperAdminPage.test.tsx` module ([link](https://github.com/medplum/medplum/pull/1814/files?diff=unified&w=0#diff-0a4ff3994d84b1699a01b19275b052f3816dffd98d534ed5f7ddb0ca79ad51b3R102-R115))
* Add the id property to the bot object in the test data in the `cron.test.ts` module ([link](https://github.com/medplum/medplum/pull/1814/files?diff=unified&w=0#diff-99c8f6f5930b08b0e6c7ca9a0aae07f2b68cfdbc13f11a156ef060bdbc7a4d58R138))

### CODE REVIEW

File: packages/server/src/workers/cron.ts [hunk 1]
```diff
@@ -225,13 +225,11 @@ export async function execBot(job: Job<CronJobData>): Promise<void> {
...
-  const previousJobs = await queue?.getRepeatableJobs();
-  previousJobs?.map(async (p) => {
-    if (p.key.includes(botId)) {
-      const keyToRemove = p.key;
-      await queue?.removeRepeatableByKey(keyToRemove);
-      logger.debug(`Found a previous job for bot ${botId}, updating...`);
-    }
-  });
+export async function removeBullMQJobByKey(botId: string): Promise<void> {
+  const previousJobs = (await queue?.getRepeatableJobs())?.filter((p) => p.id === botId) ?? [];
```
Comment (tonylee80) [inline packages/server/src/workers/cron.ts:229]: Sample  output of `console.log(await queue?.getRepeatableJobs());`

```
[
  {
    key: 'CronJobData:9f13c788-5b32-4d86-a878-758f2f764e70:::0 */12 * * *',
    name: 'CronJobData',
    id: '9f13c788-5b32-4d86-a878-758f2f764e70',
    endDate: null,
    tz: null,
    pattern: '0 */12 * * *',
    next: 1681282800000
  }
]
```

File: packages/server/src/workers/cron.test.ts [hunk 2]
```diff
@@ -135,6 +135,7 @@ describe('Cron Worker', () => {
     queue.getRepeatableJobs.mockImplementation(() => [
       {
         key: `CronJobData:${bot.id}:::* * * * *`,
+        id: bot.id,
```
Comment (tonylee80) [inline packages/server/src/workers/cron.test.ts:138]: refer to https://github.com/medplum/medplum/pull/1814/files#r1163664216

File: packages/server/src/workers/cron.ts (hunk 1 above)
Comment (jamestouri) [reply packages/server/src/workers/cron.ts:229]: Oh nice! This is way better

File: packages/server/src/workers/cron.ts [hunk 3]
```diff
@@ -225,13 +225,11 @@ export async function execBot(job: Job<CronJobData>): Promise<void> {
...
+
+  for (const p of previousJobs) {
```
Comment (jamestouri) [inline packages/server/src/workers/cron.ts:231]: Hopefully the length of previousJobs is always 1, otherwise we did something wrong

**Review by codyebberson — APPROVED** [review]
Nice!

**Pull request merged**